includes/WPAdmin/Listeners/InstallerListener.php

```diff
@@ -58,4 +62,33 @@ public function enqueue_installer_script() {
 			wp_enqueue_style( 'nfd-installer-enqueue' );
 		}
 	}
+
+	/**
+	 * Listens for premium plugin activation.
+	 *
+	 * @return void
+	 */
+	private function listen_for_premium_plugin_activation() {
+		$pls_utility = new PLSUtility();
+
+		// Retrieve the license data (decrypted) from the option
+		$license_data_store = $pls_utility->retrieve_license_data();
+
+		if ( ! $license_data_store || empty( $license_data_store ) ) {
+			return;
+		}
+
+		// Loop through each plugin in the license data and hook to its activation using basename
+		foreach ( $license_data_store as $plugin_slug => $license_data ) {
+			if ( isset( $license_data['basename'] ) ) {
+				$basename = $license_data['basename'];
+				add_action(
+					'activate_' . $basename,
+					function () use ( $plugin_slug, $pls_utility ) {
+						$pls_utility->activate_license( $plugin_slug );
```

Comment:
Just a doubt, when this hook runs, activation will fail but then activate_license completes, does it then activate the plugin as well?

Reply:
Changed to `activated_plugin`